### ISSUE TITLE
Support Elastic Horovod on Spark

### ISSUE THREAD

**Issue by tgaddair** [horovod/horovod#1874]
Recent work done to add Gloo support for Horovod on Spark (#1807) makes it compatible with the Elastic mode for Horovod (#1849).  We should integrate these systems.

**Comment by EnricoMi** [comment]
Working on it.

**Comment by EnricoMi** [comment]
Implemented in #1909 and #1956.

**Comment by EnricoMi** [comment]
I think this is in a working state. All that is left to do is add some example to the `elastic.rst`: https://github.com/horovod/horovod/blob/master/docs/elastic.rst#running-on-spark

**Comment by EnricoMi** [comment]
@tgaddair btw, github does not show any of the examples in `elastic.rst`, e.g. https://github.com/horovod/horovod/blob/master/docs/elastic.rst#elastic-tensorflow, it probably does not support the `:emphasize-lines: ...` and then ignores the whole block. I can see it works on https://horovod.readthedocs.io/en/latest/elastic_include.html#elastic-tensorflow.

**Comment by tgaddair** [comment]
Good point @EnricoMi, the `:emphasize-lines` thing is an issue.  Since a lot of people look to GitHub for docs, I think we should make an effort to fix this.  

@sblotner, any thoughts?  Is there a way to maybe show just a normal code block on GitHub but the emphasized lines in the ReadTheDocs?

**Comment by tgaddair** [comment]
Looks like GitHub has an open issue to track this here: https://github.com/github/markup/issues/914

But no one has responded :(

**Comment by sblotner** [comment]
That's really disappointing, especially because it's affecting code blocks with the :linenos: directive too. Some (not ideal) workarounds: 
- Remove all emphasis from code blocks for now :( 
- Above the code-block:: in the .rst, link to the raw .rst file as "source" (https://raw.githubusercontent.com/horovod/horovod/master/docs/elastic.rst) so people viewing in GitHub can see the examples.
- Add the examples with the emphasize lines styling as files in https://github.com/horovod/horovod/tree/master/examples, and above the code-block:: in the .rst, link to the file as "source" so people viewing in GitHub can open them. Not ideal because then you maintain the example in 2 places.

**Comment by EnricoMi** [comment]
Good suggestions! Another option:

- Above the code-block, link the respective section on horovod.readthedocs.io, e.g. https://horovod.readthedocs.io/en/latest/elastic_include.html#elastic-tensorflow.

**Comment by sblotner** [comment]
@EnricoMi I think I like that approach best, because then there's a single source of truth and everyone can see the highlighting. The language could be like, "Don't see an example below? Go to https://horovod.readthedocs.io/en/latest/elastic_include.html#elastic-tensorflow."
